include/fiction/algorithms/physical_design/graph_oriented_layout_design.hpp

```diff
@@ -80,6 +106,38 @@ struct graph_oriented_layout_design_params
      * crossing-free wiring is found. Defaults to false.
      */
     bool planar = false;
+    /**
+     * The `cost_objective` enum defines various cost objectives that can be used in the graph-oriented layout design
+     * process. Each cost objective represents a different metric used to expand a vertex in the search space graph.
+     */
+    enum cost_objective : std::uint8_t
+    {
+        /**
+         * AREA: Optimizes for the total area of the layout, aiming to minimize the space required for the design.
+         */
+        AREA,
+        /**
+         * WIRES: Optimizes for the number of wire segments in the layout, reducing the delay and increasing throughput.
+         */
+        WIRES,
+        /**
+         * CROSSINGS: Optimizes for the number of wire crossings in the layout.
+         */
+        CROSSINGS,
+        /**
+         * ACP (Area-Crossings Product): Optimizes for a combination of layout area and the number of crossings.
+         */
+        ACP,
+        /**
+         * CUSTOM: Allows for a user-defined cost objective, enabling optimization based on specific criteria outside
+         * the predefined options.
+         */
+        CUSTOM
```

Comment:
I agree that it's not suitable for the CLI. 

However, I have yet to understand how your custom functions work exactly.

Reply:
It can be passed as a function to the algorithm call:

```
const std::function<uint64_t(const gate_layout&)> custom_cost_objective = [](const gate_layout& layout) -> uint64_t
  {
      return (layout.num_wires() * 2) + layout.num_crossings();
  };

const auto layout = graph_oriented_layout_design<gate_layout>(ntk, params, &stats, custom_cost_objective);
```

Reply:
Very cool. Can you make sure to add a `@param` docstring for that parameter?